src/components/trashlookup.jsx

```diff
@@ -1,156 +1,121 @@
+      : moment()
+          .add(12 - today, 'd')
+          .format(dateFormat);
+  }
 
-  getSearchResult = () => {
-    const searchQuery = this.state.searchQuery;
-    const issearch = this.state.issearch;
-    let filtered = ""; // PostData;
-    // console.log("value of searchQuery " + searchQuery);
-    //console.log("value of isearch is " + issearch);
-    if (issearch === 1) {
-      console.log("searchclicked - false ");
-      if (searchQuery)
-        filtered = PostData.filter(m =>
-          m.address1.toLowerCase().startsWith(searchQuery.toLowerCase())
-        );
-    } else {
-      if (issearch === 2) console.log("searchclicked - true ");
-      filtered = PostData.filter(m =>
-        m.address1.toLowerCase().startsWith(searchQuery.toLowerCase())
-      );
-    }
+  recylceNextCollectionDate() {
+    const { dateFormat, dayOfWeek } = this.state;
+    const today = moment().day();
+    return today >= dayOfWeek.sunday && today < dayOfWeek.friday
+      ? moment()
+          .add(5 - today, 'd')
+          .format(dateFormat)
+      : moment()
+          .add(6, 'd')
+          .format(dateFormat);
+  }
 
-    console.log(filtered.length);
-    console.log(filtered);
-    return { totalCount: filtered.length, data: filtered, issearch: issearch };
-  };
+  trashNextCollectionDate() {
+    const { dateFormat, dayOfWeek } = this.state;
+    const today = moment().day();
 
-  render() {
-    const { totalCount, data: data, issearch } = this.getSearchResult();
+    return today >= dayOfWeek.monday && today < dayOfWeek.saturday
+      ? moment()
+          .add(6 - today, 'd')
+          .format(dateFormat)
+      : today === dayOfWeek.saturday
+      ? moment()
+          .add(2, 'd')
+          .format(dateFormat)
+      : moment()
+          .add(1, 'd')
+          .format(dateFormat);
+  }
 
-    const searchQuery = this.state.addresses;
-    const searchValue = this.state.searchQuery;
+  render() {
+    const { data } = this.addressData();
+    const isCollectionScheduleTextHidden = this.state.isCollectionScheduleTextHidden;
+    const isAutoTextHidden = this.state.isAutoTextHidden;
     return (
       <React.Fragment>
-        <p>Showing {totalCount} address in the database.</p>
-        <SearchBox
-          value={searchValue}
-          onChange={this.handleSearch}
-          onClick={this.handleSearchClick}
-        />
+        <h6>Find Your Collection Schedule.</h6>
+        <p>First paragraph shown here</p>
+        <p>Second paragraph shown here</p>
         <div className="row">
-          {issearch === 1 && totalCount > 0 ? (
-            <div className="col-5">
-              <ListGroup
-                items={data}
-                selectedItem={this.state.selectedAddress}
-                onItemSelect={this.handleAddressSelect}
-              />
-            </div>
-          ) : (
-            ""
-          )}
+          <div className="col-5">{!isAutoTextHidden && this.renderList(data)}</div>
```

Comment:
in most cases where you have methods that start with "render" should be moved to their own components. If you need something from this component they should be passed as props.